src/multi_physics/QED/src/physics/breit_wheeler/breit_wheeler_engine_tables.hpp

```diff
@@ -0,0 +1,828 @@
+                }
+
+                m_params = serialization::get_out<
+                    pair_prod_lookup_table_params<RealType>>(it_raw_data);
+                m_table = containers::equispaced_2d_table<
+                    RealType, VectorType>{std::vector<char>(it_raw_data,
+                    raw_data.end())};
+
+                m_init_flag = true;
+            };
+
+            /**
+            * Operator==
+            *
+            * @param[in] rhs a structure of the same type
+            *
+            * @return true if rhs is equal to *this. false otherwise
+            */
+            PXRMP_INTERNAL_GPU_DECORATOR PXRMP_INTERNAL_FORCE_INLINE_DECORATOR
+            bool operator== (
+                const pair_prod_lookup_table<
+                    RealType, VectorType> &rhs) const
+            {
+                return
+                    (m_params == rhs.m_params) &&
+                    (m_init_flag == rhs.m_init_flag) &&
+                    (m_table == rhs.m_table);
+            }
+
+            /*
+            * Returns a table view for the current table
+            * (i.e. a table built using non-owning picsar_span
+            * vectors). A view_type is very light weight and can
+            * be passed by copy to functions and GPU kernels.
+            * Indeed it contains non-owning pointers to the data
+            * held by the original table.
+            * The method is not designed to be run on GPUs.
+            *
+            * @return a table view
+            */
+            view_type get_view() const
+            {
+                if(!m_init_flag)
+                    throw std::runtime_error("Can't generate a view of an \
+                    uninitialized table");
+                const auto span = containers::picsar_span<const RealType>{
+                    static_cast<size_t>(m_params.chi_phot_how_many *
+                        m_params.frac_how_many),
+                        m_table.get_values_reference().data()};
+
+                return view_type{m_params, span};
+            }
+
+            /*
+            * Uses the lookup table to extract the chi value of
+            * one of the generated particles from a cumulative probability
+            * distribution, given the chi paramter of the photon and a
+            * random number uniformly distributed in [0,1). If chi_phot is out
+            * of table either the minimum or the maximum value is used.
+            * The method uses the lookup table to invert the equation:
+            * unf_zero_one_minus_epsi = P(chi_phot, X)
+            * where X is the ratio between chi_particle and chi_photon.
+            * Symmetry of the cumulative probability distribution function is
+            * exploited (P actually goes up to 0.5 and X goes up to 0.5).
+            * If unf_zero_one_minus_epsi < 0.5 X*chi_phot is returned,
+            * otherwise (1-X)*chi_phot is returned.
```

Comment:
```suggestion
            * If unf_zero_one_minus_epsi < 0.5 then X*chi_phot is returned.
            * Otherwise (1-X)*chi_phot is returned.
```
Maybe slightly less ambiguous this way?